Extension/YapoSubModule.cs

```diff
@@ -42,18 +41,33 @@ protected override void OnGameStart(Game game, IGameStarter gameStarterObject)
             base.OnGameStart(game, gameStarterObject);
             if (!(game.GameType is Campaign)) return;
 
-            MBObjectManager.Instance.RegisterType<TroopSorterService>("Sorter", "Sorters");
-            
-            (gameStarterObject as CampaignGameStarter)?.LoadGameTexts($"{BasePath.Name}/Modules/{Strings.MODULE_FOLDER_NAME}/{Strings.MODULE_DATA_PARTY_COUNT_STRINGS}");
+            try
+            {
+                MBObjectManager.Instance.RegisterType<TroopSorterService>("Sorter", "Sorters");
+                AddBehaviours(gameStarterObject as CampaignGameStarter);
+            } catch (Exception exception)
+            {
+                ModDebug.ShowError("Something went wrong while starting a game", "OnGameStart exception", exception);
+            }
+        }
+
+        private static void AddBehaviours(CampaignGameStarter campaignGameStarter)
+        {
+            if (campaignGameStarter == null) return;
+
+            if (YapoSettings.Instance.IsFormationPersistenceFixEnabled)
+            {
+                campaignGameStarter.AddBehavior(FixedFormationsBehaviour.Instance);
+                campaignGameStarter.LoadGameTexts($"{BasePath.Name}/Modules/{Strings.MODULE_FOLDER_NAME}/{Strings.MODULE_DATA_FORMATION_STRINGS}");
+            }
+
+            campaignGameStarter.LoadGameTexts($"{BasePath.Name}/Modules/{Strings.MODULE_FOLDER_NAME}/{Strings.MODULE_DATA_PARTY_COUNT_STRINGS}");
         }
 
         protected override void OnApplicationTick(float dt)
         {
-            if (States.PartyVmMixin == null
-                || Campaign.Current == null
-                || Campaign.Current.CurrentMenuContext != null
-                && (!Campaign.Current.CurrentMenuContext.GameMenu.IsWaitActive
-                    || Campaign.Current.TimeControlModeLock))
+            if (States.PartyVmMixin == null || Campaign.Current == null ||
+                Campaign.Current.CurrentMenuContext != null && (!Campaign.Current.CurrentMenuContext.GameMenu.IsWaitActive || Campaign.Current.TimeControlModeLock))
```

Comment:
Would be good to keep the formatting that was here before. Maybe a good idea to disable formatting for manually wrapped lines (this was a default on my fresh rider install)

Reply:
Yeah I have my wrap/chop setting at a higher line length than standard. Can you check to see if this line fits like that on your screen?
I'm not a fan of aggressively wrapped/chopped lines

Reply:
This is really dense logic that is more readable over multiple lines. Generally I would hard break lines at 120 characters. In my personal projects I even break earlier since it makes reviewing code so much easier.